lib/sda/tree.go

```diff
@@ -258,7 +259,7 @@ func (tm TreeMarshal) MakeTree(il *EntityList) (*Tree, error) {
 
 // MakeTreeFromList creates a sub-tree given an EntityList
 func (tm *TreeMarshal) MakeTreeFromList(parent *TreeNode, il *EntityList) *TreeNode {
-	idx, ent := il.Search(tm.EntityId)
+	idx, ent := il.SearchEntity(tm.EntityId)
```

Comment:
In fact the `TreeNode`-structure does have `EntityIdx`. But I'd still add a comment there saying that `EntityIdx` is not unique.